src/parsers/gpgsa.c

```diff
@@ -0,0 +1,112 @@
+#include <ctype.h>
+
+#include "../nmea/parser_types.h"
+#include "gpgsa.h"
+#include "parse.h"
+
+int
+init(nmea_parser_s *parser)
+{
+	/* Declare what sentence type to parse */
+	NMEA_PARSER_TYPE(parser, NMEA_GPGSA);
+	NMEA_PARSER_PREFIX(parser, "GPGSA");
+	return 0;
+}
+
+int
+allocate_data(nmea_parser_s *parser)
+{
+	parser->data = malloc(sizeof (nmea_gpgsa_s));
+	if (NULL == parser->data) {
+		return -1;
+	}
+
+	return 0;
+}
+
+int
+set_default(nmea_parser_s *parser)
+{
+	memset(parser->data, 0, sizeof (nmea_gpgsa_s));
+	return 0;
+}
+
+int
+free_data(nmea_s *data)
+{
+	free(data);
+	return 0;
+}
+
+int
+parse(nmea_parser_s *parser, char *value, int val_index)
+{
+	nmea_gpgsa_s *data = (nmea_gpgsa_s *) parser->data;
+
+	switch (val_index)
+  {
```

Comment:
and here as well, please replace leading spaces with tabs

Reply:
Done